python/lsst/ts/externalscripts/base_parameter_march.py

```diff
@@ -0,0 +1,461 @@
+
+        Parameters
+        ----------
+        config : `types.SimpleNamespace`
+            Script configuration, as defined by `schema`.
+        """
+        await self.configure_tcs()
+        await self.configure_camera()
+        await self.configure_ocps()
+
+        if hasattr(config, "ignore"):
+            for comp in config.ignore:
+                if comp in self.tcs.components_attr:
+                    self.log.debug(f"Ignoring TCS component {comp}.")
+                    setattr(self.tcs.check, comp, False)
+                elif comp in self.camera.components_attr:
+                    self.log.debug(f"Ignoring Camera component {comp}.")
+                    setattr(self.camera.check, comp, False)
+                else:
+                    self.log.warning(
+                        f"Component {comp} not in CSC Groups. "
+                        f"Must be one of {self.tcs.components_attr} or "
+                        f"{self.camera.components_attr}. Ignoring."
+                    )
+
+        if hasattr(config, "step_sequence"):
+            config.range = config.step_sequence[-1] - config.step_sequence[0]
+            config.n_steps = len(config.step_sequence)
+        elif hasattr(config, "range"):
+            config.step_sequence = np.linspace(
+                -config.range, config.range, config.n_steps
+            ).tolist()
+
+        if hasattr(config, "rotation_sequence"):
+            if isinstance(config.rotation_sequence, (int, float)):
+                config.rotation_sequence = [config.rotation_sequence] * config.n_steps
+            elif isinstance(config.rotation_sequence, list):
+                if len(config.rotation_sequence) != config.n_steps:
+                    raise ValueError(
+                        f"rotation_sequence length {len(config.rotation_sequence)} "
+                        f"does not match n_steps {config.n_steps}."
+                    )
+            else:
+                raise TypeError("rotation_sequence must be either a number or a list.")
+
+        self.config = config
+        if hasattr(config, "dofs"):
+            self.dofs = np.array(config.dofs)
+        elif hasattr(config, "dof_index"):
+            self.dofs = np.zeros(50)
+            self.dofs[config.dof_index] = 1.0
+
+        await super().configure(config=config)
+
+    def set_metadata(self, metadata: salobj.type_hints.BaseMsgType) -> None:
+        """Sets script metadata.
+
+        Parameters
+        ----------
+        metadata : `salobj.type_hints.BaseMsgType`
+            Script metadata topic.
+        """
+        raise NotImplementedError()
+
+    async def assert_feasibility(self) -> None:
+        """Verify that the telescope and camera are in a feasible state to
+        execute the script.
+        """
+        await asyncio.gather(
+            self.tcs.assert_all_enabled(), self.camera.assert_all_enabled()
+        )
+
+    async def format_values(self, offset_values: np.ndarray) -> np.ndarray:
```

Comment:
This method is not using any attribute from the class. Maybe make it a `@staticmethod`?

Reply:
you need to remove the `self` argument for a static method.